packages/scripts/scripts/check-licenses.js

```diff
@@ -193,6 +225,11 @@ modules.forEach( ( path ) => {
 		);
 	let licenseType = typeof license === 'object' ? license.type : license;
 
+	// Check if the license we've detected is telling us to look in the license file, instead.
+	if ( licenseType && licenseFiles.find( ( licenseFile ) => licenseType.indexOf( licenseFile ) >= 0 ) ) {
```

Comment:
nit: This is a good place to use `String#includes` which reads a bit better than an index comparison.